language/stdlib/modules/transaction_fee.move

```diff
@@ -1,105 +1,87 @@
 address 0x0 {
 
 module TransactionFee {
-    use 0x0::LibraAccount;
-    use 0x0::LibraSystem;
     use 0x0::Signer;
     use 0x0::Transaction;
+    use 0x0::Coin1::T as Coin1;
+    use 0x0::Coin2::T as Coin2;
+    use 0x0::LBR::{Self, T as LBR};
+    use 0x0::Libra::{Self, T as Libra};
 
-    ///////////////////////////////////////////////////////////////////////////
-    // Transaction Fee Distribution
-    ///////////////////////////////////////////////////////////////////////////
-    // Implements a basic transaction fee distribution logic.
-    //
-    // We have made a couple design decisions here that are worth noting:
-    //  * We pay out once per-block for now.
-    //    TODO: Once we have a better on-chain representation of
-    //          epochs this should be changed over to be once per-epoch.
-    //  * Sometimes the number of validators does not evenly divide the transaction fees to be
-    //    distributed. In such cases the remainder ("dust") is left in the transaction fees pot and
-    //    these remaining fees will be included in the calculations for the transaction fee
-    //    distribution in the next epoch. This distribution strategy is meant to in part minimize the
-    //    benefit of being the first validator in the validator set.
+    resource struct FeePaymentCapability { }
 
-    resource struct TransactionFees {
-        fee_withdrawal_capability: LibraAccount::WithdrawalCapability,
+    resource struct CollectedFees<CoinType> {
+        preburn: Libra::Preburn<CoinType>,
     }
 
-    // Initialize the transaction fee distribution module. We keep track of the last paid block
-    // height in order to ensure that we don't try to pay more than once per-block. We also
-    // encapsulate the withdrawal capability to the transaction fee account so that we can withdraw
-    // the fees from this account from block metadata transactions.
-    public fun initialize_transaction_fees(fee_account: &signer) {
+    resource struct LBRIdent<LBRCoinType> { coin: Libra<LBRCoinType> }
+
+    // Called in genesis. Once for each fiat currency.
+    public fun add_transaction_fee<CoinType>(fee_account: &signer, preburn: Libra::Preburn<CoinType>) {
```

Comment:
I find this name a bit confusing; it sounds like it might be paying a fee. Maybe something like `add_currency_type`?

Reply:
How does `add_transaction_fee_currency` sound?

Reply:
`transaction_fee` part sounds a bit redundant given the module name, but not picky :)